planet/src/main/webapp/js/itinerary.js

```diff
@@ -49,19 +63,67 @@ function renderStartingLocation() {
     }
 }
 
+function renderListOptions() {
+    const userId = firebase.auth().currentUser.uid;
+    const eventListRef = database.ref('events/' + userId);
+    const selectListElement = document.getElementById('list-options');
+    // Show the option "current list" regardless of what's stored in the database
+    selectListElement.innerHTML = '<option value="currentList">Current List </option>';
+    eventListRef.once('value', function(eventsSnapshot) {
+        eventsSnapshot.forEach(function(childListSnapshot) {
+            let childKey = childListSnapshot.key;
+            if (childKey !== 'currentList') {
+                let optionElement = document.createElement('option');
+                optionElement.value = childKey;
+                optionElement.innerText = childKey;
+                selectListElement.add(optionElement);
+                // Update the selected value
+                if (childKey === sessionStorage.getItem('listName')) {        
+                    optionElement.selected = true;
+                }
+            }
+        });
+        // Initialize the select tag again since options are dynamically loaded
+        M.FormSelect.init(selectListElement, {});
+    });
+}
+
+function handleListOptionChange() {
+    const listName = document.getElementById ('list-options').value;
+    sessionStorage.setItem('listName', listName);
+    // You can only save currentList as other lists
+    if (listName !== 'currentList') {
+        document.getElementById('save-events-button').style.display = 'none';
+    }else{
+        document.getElementById('save-events-button').style.display = 'inline-block';
```

Comment:
How do you save a  list as itself? E.g. I have a list called "test", and I want to save changes to it.

Reply:
Changes will be automatically saved if users are on other lists (just like how currentList is automatically updated). The save button is for exporting currentList. Hope that makes sense!